quinn/src/endpoint.rs

```diff
@@ -253,39 +253,43 @@ where
 {
     fn drive_recv(&mut self, cx: &mut Context, now: Instant) -> Result<bool, io::Error> {
         let mut recvd = 0;
+        let mut metas = [RecvMeta::default(); BATCH_SIZE];
+        let mut iovs = [IoSliceMut::new(&mut [])];
+        for (iov, buf) in iovs
+            .iter_mut()
+            .zip(self.recv_buf.chunks_mut(self.recv_buf.len() / BATCH_SIZE))
+        {
+            *iov = IoSliceMut::new(buf);
+        }
```

Comment:
This confuses me. Why do you instantiate a one-element `IoSliceMut` array first, then iterate over it and overwrite it?

Reply:
Good catch, this wasn't intended, and was rendering the whole PR moot. Still no drastic change in perf though, even after manually verifying that multiple packets are being received per batch...